### ISSUE TITLE
Custom CORS configuration

### ISSUE THREAD

**Issue by djannot** [solo-io/gloo-portal-issues#106]
**Is your feature request related to a problem? Please describe.**
The VirtualService generated by Gloo Portal includes a CORS configuration but:
- the `content-type` isn't part of the `allowHeaders`
- the `allowOrigin` can't be customized

**Describe the solution you'd like**
We need to either provide a way for the user to provide its own CORS configuration or at least some entries they want to add to the `allowHeaders` and `allowOrigin` arrays.

**Comment by marcogschmidt** [comment]
Adding some context and initial thoughts. We currently set CORS configuration on the Gloo Edge `VirtualService` that is generated for an `Environment` only if any of the API product versions included in that `Environment` are published in one or more `Portal` resources. This is done in order for the "Try it out" feature in the interactive docs for those APIs to work in portal applications (as the requests sent from there are cross-origin). Specifically, we set the following CORS configuration:

```yaml
# The domains of the relevant portals (each with both http and https schemes)
allowOrigin: [] 
# Regular expressions for the domains of the relevant portals 
# that contain DNS wildcards (each with both http and https schemes)
allowOriginRegex: []
allowCredentials: true
allowHeaders:
- api-key
- Authorization
```

Although we allow users to set a wide array of options on the generated `VirtualService`, we currently do not allow setting custom CORS policies in not to conflict with these automatically added ones (see docs [here](https://docs.solo.io/gloo-portal/latest/reference/api/environment/#portal.gloo.solo.io.EnvironmentSpec.GatewayConfig)). I think the best solution would be to relax this constraint and allow users to provide their custom options.

**Comment by marcogschmidt** [comment]
We have to decide whether we want the `disableAutoCors` in the initial version of this feature. If so, then we need to update the admin UI as well to allow users to set this flag.

**Comment by marcogschmidt** [comment]
After discussing this internally we decided to implement the following behavior:
- if the user did not provide any CORS configuration on the environment `virtualHostOptions`, then autogenerate the CORS configuration to make the "try it out" feature work
- if the user provided CORS configuration on the environment `virtualHostOptions`, then don't add anything there. It will be up to the user to make sure CORS is configured correctly to work with the "try it out" feature
- if the user provides CORS configuration on the APIProduct routes, then just propagate that to the route options. The route-level config can override the one on the virtual host (e.g. if `allowedOrigin` is defined in both, the route config wins), so it will be up to the user to make sure the final CORS configuration is correct.

**Comment by marcogschmidt** [comment]
This change has been merged and will be released with Gloo Portal `v1.1.0-beta7`.
